cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java

```diff
@@ -374,8 +374,7 @@ private Integer getResolution(String resolution, Long start, Long end) {
 
   private MetricQueryResult executeQuery(QueryRequest queryRequest) {
     try {
-
-      if (queryRequest.getMetrics() == null || queryRequest.getMetrics().size() == 0) {
+      if (queryRequest.getMetrics().size() == 0) {
```

Comment:
OK. IllegalArgumentException is not really a good indicator for a bad request, because many internal errors are also IllegalArgumentException (whenever Preconditions.checkArgument throws). But for now I will say it's ok.